crates/core/component/dex/src/component/router/fill_route.rs

```diff
@@ -351,15 +350,15 @@ impl<S: StateRead + StateWrite> Frontier<S> {
 
         // Record a trace of the execution along the current route,
         // starting with all-zero amounts.
-        let trace: Vec<Value> = std::iter::once(Value {
+        let trace: Vec<Vec<Value>> = vec![std::iter::once(Value {
             amount: 0u64.into(),
             asset_id: pairs.first().unwrap().start,
         })
         .chain(pairs.iter().map(|pair| Value {
             amount: 0u64.into(),
             asset_id: pair.end,
         }))
-        .collect();
+        .collect()];
```

Comment:
This will insert an all-zeros execution at the beginning, instead, it should be `let trace = Vec::new()` to represent an empty set of traces.